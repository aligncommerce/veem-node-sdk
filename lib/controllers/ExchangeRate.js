import isArray from 'lodash/isArray'
import { POST } from 'constants/httpMethods'

class ExchangeRate {
  constructor (sdk) {
    this.sdk = sdk
  }

  quote (quotes, callback) {
    const isBatch = isArray(quotes)

    const path = isBatch
      ? '/veem/v1.1/exchangerates/quotes/batch'
      : '/veem/v1.1/exchangerates/quotes'

    return this.sdk.apiClient.call({
      method: POST,
      path,
      payload: quotes,
    }, callback)
  }
}

export default ExchangeRate
