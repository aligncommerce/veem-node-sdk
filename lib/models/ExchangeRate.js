import ExchangeRateRequest from './ExchangeRateRequest'
import ExchangeRateResponse from './ExchangeRateResponse'

class ExchangeRate {}

ExchangeRate.request = ExchangeRateRequest
ExchangeRate.response = ExchangeRateResponse

export default ExchangeRate
