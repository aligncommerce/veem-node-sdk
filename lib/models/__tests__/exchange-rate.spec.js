import ExchangeRate from 'models/exchange-rate'

describe('ExchangeRate Model', () => {
  describe('request', () => {
    it('should match the snapshot', () => {
      expect(ExchangeRate.request.schema).toMatchSnapshot()
    })

    it('should define a valid schema', () => {
      expect(ExchangeRate.request.schema).to.be.validJsonSchema
    })
  })

  describe('response', () => {
    it('should match the snapshot', () => {
      expect(ExchangeRate.response.schema).toMatchSnapshot()
    })

    it('should define a valid schema', () => {
      expect(ExchangeRate.response.schema).to.be.validJsonSchema
    })
  })
})
