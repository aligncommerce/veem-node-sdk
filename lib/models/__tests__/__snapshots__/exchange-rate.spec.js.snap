// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExchangeRate Model request should match the snapshot 1`] = `
Object {
  "properties": Object {
    "fromAmount": Object {
      "type": "string",
    },
    "fromCurrency": Object {
      "type": "string",
    },
    "recipientAccountEmail": Object {
      "type": "string",
    },
    "toAmount": Object {
      "type": "string",
    },
    "toCountry": Object {
      "type": "string",
    },
    "toCurrency": Object {
      "type": "string",
    },
  },
  "title": "ContactRequest",
  "type": "object",
}
`;

exports[`ExchangeRate Model response should match the snapshot 1`] = `
Object {
  "properties": Object {
    "expiry": Object {
      "type": "number",
    },
    "fromAmount": Object {
      "type": "number",
    },
    "fromCurrency": Object {
      "type": "string",
    },
    "id": Object {
      "type": "string",
    },
    "rate": Object {
      "type": "number?",
    },
    "timeCreated": Object {
      "type": "string?",
    },
    "toAmount": Object {
      "type": "number",
    },
    "toCurrency": Object {
      "type": "string",
    },
  },
  "title": "ExchangeRateResponse",
  "type": "object",
}
`;
